bench/lib/FilterJobBench.re

```diff
@@ -26,17 +26,32 @@ let largeAmountOfItems =
     )
   );
 
-let job =
+let smallerNumberOfItems =
+  List.init(10000, i =>
+    createItem(
+      "Some item with a long name but with index " ++ string_of_int(i),
+    )
+  );
+
+let getJob = (items) =>
   MenuFilterJob.create()
   |> Job.map(MenuFilterJob.updateQuery("item 1"))
-  |> Job.map(MenuFilterJob.addItems(largeAmountOfItems))
-  |> Job.map(MenuFilterJob.addItems(largeAmountOfItems))
-  |> Job.map(MenuFilterJob.addItems(largeAmountOfItems))
-  |> Job.map(MenuFilterJob.addItems(largeAmountOfItems))
-  |> Job.map(MenuFilterJob.addItems(largeAmountOfItems));
+  |> Job.map(MenuFilterJob.addItems(items))
+  |> Job.map(MenuFilterJob.addItems(items))
+  |> Job.map(MenuFilterJob.addItems(items))
+  |> Job.map(MenuFilterJob.addItems(items))
+  |> Job.map(MenuFilterJob.addItems(items));
 
+let job = getJob(largeAmountOfItems);
 let runJob = () => Job.doWork(job) |> (ignore: Job.t(_) => unit);
 
+let completeJob = () => {
+  let job = ref(getJob(smallerNumberOfItems))
+  while (! Job.isComplete(job^)) {
+    job := Job.doWork(job^);
+  }
+}
+
```

Comment:
Added this to more easily benchmark changes to the full (I think this is the full matching?) matching, rather than a single run of `doWork`.

Obviously knowing how fast `doWork` is useful but its not much help tuning the full performance (tuning limits or the pre-fuzzy step), since turning the limits up makes 1 take longer but you do less and removing the fuzzy makes it quicker etc.

Reply:
Makes sense!